stacks-doctor/src/commands/reorgs.rs

```diff
@@ -0,0 +1,103 @@
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use rusqlite::OpenFlags;
+use serde::Serialize;
+use serde_json::to_string_pretty;
+
+use crate::cli::{BlocksArgs, Network};
+
+#[derive(Serialize)]
+struct Item {
+    burn_height: i64,
+    stacks_block_id: String,
+    miner: String,
+    stacks_height: i64,
+    reorg_depth: i64,
+}
+
+#[derive(Serialize)]
+struct Response {
+    message: String,
+    max_reorg_depth: i64,
+    max_reorg_blocks_ago: usize,
+    data: Vec<Item>,
+}
+
+pub fn reorgs(network: Network, db_dir: &Path, args: &BlocksArgs) -> Result<()> {
+    let mode = match network {
+        Network::Mainnet => "mainnet/",
+        Network::Testnet => "xenon/",
+    };
+    let db_file = db_dir.join(mode).join("chainstate/vm/index.sqlite");
+    let conn = rusqlite::Connection::open_with_flags(db_file, OpenFlags::SQLITE_OPEN_READ_ONLY)
+        .context("Could not open database connection")?;
+
+    let mut statement = conn
+        .prepare(
+            r#"
+            SELECT
+               block_headers.burn_header_height as burn_height,
+               b.index_block_hash as index_block_hash,
+               payments.address as miner,
+               block_headers.block_height as stacks_height
+            FROM staging_blocks as b
+            INNER JOIN block_headers ON block_headers.index_block_hash = b.index_block_hash
+            INNER JOIN payments ON payments.index_block_hash = b.index_block_hash
+            WHERE payments.miner = 1 GROUP BY block_headers.burn_header_height
+            ORDER BY burn_height DESC
+            LIMIT ?1;
+        "#,
+        )
+        .context("Could not prepare SQL statement")?;
+
+    let mut rows = statement
+        .query::<[i64; 1]>([args.blocks as i64 + 1])
+        .context("Could not execute query")?;
+
+    let mut data = vec![];
```

Comment:
Nit: Sergey likes to avoid macros when they don't provide much extra value. Here the constructor call reads better.
```suggestion
    let mut data = Vec::new();
```

Reply:
Okay that's fair, will do.